lib/fibonacci.rb

```diff
@@ -1,8 +1,26 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 # Hint, you may want a recursive helper method
-def fibonacci(n)
- 
+def fibonacci(n) 
```

Comment:
This works, but you're keeping the entire list of fibonacci numbers from 0 to n  for the entire set of recursive calls.  You only need to keep the last 2 numbers.